modules/ppcp-axo/src/FrontendLoggerEndpoint.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * The endpoint to log entries from frontend.
+ *
+ * @package WooCommerce\PayPalCommerce\Button\Endpoint
+ */
+
+declare(strict_types=1);
+
+namespace WooCommerce\PayPalCommerce\Axo;
+
+use Exception;
+use Psr\Log\LoggerInterface;
+use WooCommerce\PayPalCommerce\Button\Endpoint\EndpointInterface;
+use WooCommerce\PayPalCommerce\Button\Endpoint\RequestData;
+
+/**
+ * Class FrontendLoggerEndpoint
+ */
+class FrontendLoggerEndpoint implements EndpointInterface {
+
+	const ENDPOINT = 'ppc-frontend-logger';
+
+	/**
+	 * The request data helper.
+	 *
+	 * @var RequestData
+	 */
+	private $request_data;
+
+	/**
+	 * The logger.
+	 *
+	 * @var LoggerInterface
+	 */
+	private $logger;
+
+	/**
+	 * FrontendLoggerEndpoint constructor.
+	 *
+	 * @param RequestData     $request_data The request data helper.
+	 * @param LoggerInterface $logger The logger.
+	 */
+	public function __construct( RequestData $request_data, LoggerInterface $logger ) {
+		$this->request_data = $request_data;
+		$this->logger       = $logger;
+	}
+
+	/**
+	 * Returns the nonce.
+	 *
+	 * @return string
+	 */
+	public static function nonce(): string {
+		return self::ENDPOINT;
+	}
+
+	/**
+	 * Handles the request.
+	 *
+	 * @return bool
+	 * @throws Exception On Error.
+	 */
+	public function handle_request(): bool {
```

Comment:
Even though this is just a logger, I'd recommend we implement nonce verification for all our endpoints. Like:

https://github.com/woocommerce/woocommerce-paypal-payments/blob/ed3a48e520780df5500b989d4eba3af7ccbe72b1/modules/ppcp-paylater-configurator/src/Endpoint/GetConfig.php#L82

Reply:
> Even though this is just a logger, I'd recommend we implement nonce verification for all our endpoints. Like:
> 
> https://github.com/woocommerce/woocommerce-paypal-payments/blob/ed3a48e520780df5500b989d4eba3af7ccbe72b1/modules/ppcp-paylater-configurator/src/Endpoint/GetConfig.php#L82

[Nonce verification](https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-button/src/Endpoint/RequestData.php#L47) and [input sanitization](https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-button/src/Endpoint/RequestData.php#L61) is already implemented in `RequestData::read_request`.

What we could do to avoid confusion and code duplication is to extract that logic out of buttons module to a shared one, or at least move it to the api client module.